File: solutions/typescript/src/0004.ts
function findMedianSortedArrays(nums1: number[], nums2: number[]): number {
    const merged = [...nums1, ...nums2].sort((a, b) => a - b);
    const middle = Math.floor(merged.length / 2);
    if (merged.length % 2 === 1) {
        return merged[middle];
    } else {
        return (merged[middle - 1] + merged[middle]) / 2;
    }
}

export default findMedianSortedArrays;

// function findMedianSortedArrays(nums1: number[], nums2: number[]): number {
//     let middle = 0;
//     const list = [...nums1, ...nums2].sort((a,b) => a - b);
//     if (list.length % 2 === 1) {
//         middle = list[Math.floor(list.length / 2)]
//     } else {
//         middle = (list[Math.floor(list.length / 2)-1] +  list[Math.floor(list.length / 2)]) / 2
//     }

//     return middle;
// };

File: solutions/go/0004.go
package main

import (
	"sort"
)
func findMedianSortedArrays(nums1 []int, nums2 []int) float64 {
	// 合并两个数组
	merged := append(nums1, nums2...)
	
	// 对合并后的数组进行排序
	sort.Ints(merged)
	
	// 计算中位数
	length := len(merged)
	middle := length / 2
	
	if length%2 == 1 {
		return float64(merged[middle])
	} else {
		return float64(merged[middle-1]+merged[middle]) / 2.0
	}
}

func main() {
	// 测试用例
	nums1 := []int{1, 3}
	nums2 := []int{2}
	result := findMedianSortedArrays(nums1, nums2)
	println(result)
}

